### ISSUE TITLE
Basic short answer exercise layout

### ISSUE THREAD

**Issue by ramonziai** [isaac-project/h5p-short-answers#2]
To get started in the direction of a functional short answer interface, let's first design a basic (mockup) interface that looks like the real thing, but doesn't yet provide the interaction. This will let us get to know how H5P content creation and integration works with TypeScript.

I suggest we start from a minimal codebase (on a different branch than master) with this, since much of the Advanced Blanks code might get in the way of how we want to do things. We can take over the node.js and TypeScript setup from the current master and also later draw inspiration (and code) from Advanced Blanks.

The task would be to

* come up with a semantics.json that allows for specifying 1) a reading text, 2) one or more questions about the text, 3) target answers for the questions (not shown) and 4) one student answer field per question.
* provide the TypeScript code to glue these things together and get them to show up. https://github.com/h5p/h5p-question (available as H5P library H5P.Question) might be helpful here, as it seems to provide some functionality common to question-like H5P content types.
* if the above work, provide some minimal styling to make things look a little smoother.

**Comment by edemattos** [comment]
@ramonziai Can you pull/clone the repo (development branch) into the files/h5p/development folder you've configured for Drupal, and try to add a new content instance within the browser at http://localhost:8080? It is currently named "Basic Short Answers", and you should be able to add a text (although there seems to be a character limit) and multiple FITB questions by selecting "add text block", which I think should address objectives 1-4.

**Comment by ramonziai** [comment]
@edemattos Thanks, I just tried it out and it's good to see things are going where we want them to. Two things would be good to have, though:

1. The reading text seems to be capped at 260 characters, which is about a paragraph maximum. Could you extend that (we definitely need about 5-10 times that size).
2. Your data model of questions, target and student answers is currently based on H5P.Blanks. While it's possible to realize a short answer task within that model, the two task types are actually quite different both in presentation and types of possible answers. I looked at the content.json created by Drupal for the example task I made, and it looks like there is i) no separation between HTML markup and content in the questions, and ii) no way to refer to a particular target answer, they effectively need to be parsed by looking for *bla* parts and then separating alternatives.
Long story short: Could you implement something that looks more like this: https://unitc-my.sharepoint.com/:i:/g/personal/nnszi01_cloud_uni-tuebingen_de/EWGGLBV3YDlBlYjW7nTKl_wBquN_AQCJSuQAGL9U-8T_4w?e=xpGtjm and also has a data structure like "questions: [{"id":"question123","prompt":"What is the meaning of life?", "targetAnswers":["42","forty-two"]}]" so presentation and data are more clearly separated?

**Comment by edemattos** [comment]
@ramonziai I received a prompt response on the forum that led to a nice (JavaScript) boilerplate for the Question type you proposed. I've added the skeleton to keep you in the loop while I tinker around with it.

**Comment by edemattos** [comment]
@ramonziai I updated `semantics.json` so that passage text/multimedia are both optional meaning if you want to use plain text or an image (as in the example you referred to), either can be done without requiring the other. Having both fields optional also leaves open the possibility for tasks that are not based on a passage.

Modifying `semantics.json` does not require changing any JavaScript so if you want to play around with it, you will see the changes immediately reflected in the editor on the add/edit content page.

I've made some progress with the display (JavaScript) but so far I am mostly hardcoding stuff. I was able to get the easy (plain) text to display but I still need to figure out how to incorporate the blanks for user input.

**Comment by ramonziai** [comment]
b0e3d72 makes the passage show up as textarea in the editor so it's easier to paste and work on a longer text.

**Comment by ramonziai** [comment]
Looking good so far! Closing this issue for now.
